### PR TITLE
Make modules APP DATA DOMAIN -> Create new module domain

### PR DESCRIPTION
**Pull request by Nataniel-Antosik**

### CODE REVIEW

File: domain/build.gradle [hunk 1]
```diff
@@ -0,0 +1,31 @@
+plugins {
```
Comment (MarekMacko) [inline domain/build.gradle:1]: Domain module should be independent from the platform (Android in this case) it has to be only a kotlin module without any knowledge about Android dependency

**Review by MarekMacko — CHANGES_REQUESTED** [review]

File: domain/build.gradle (hunk 1 above)
Comment (Nataniel-Antosik) [reply domain/build.gradle:1]: I changed it and now I created the module `Java or Kotlin Library` type

File: domain/build.gradle [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    id 'java-library'
```
Comment (MarekMacko) [inline domain/build.gradle]: Is it really a "java" library? Probably you want to create a kotlin library :) Try to apply only kotlin here, there is no need for any java support

**Review by MarekMacko — CHANGES_REQUESTED** [review]

File: domain/build.gradle (hunk 2 above)
Comment (Nataniel-Antosik) [reply domain/build.gradle]: Ok, this id was added default when I created the module and I deleted it :D

File: domain/build.gradle [hunk 3]
```diff
@@ -0,0 +1,8 @@
+plugins {
+    id 'org.jetbrains.kotlin.jvm'
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
```
Comment (MarekMacko) [inline domain/build.gradle]: Is this needed?

**Review by MarekMacko — CHANGES_REQUESTED** [review]

File: domain/build.gradle (hunk 3 above)
Comment (Nataniel-Antosik) [reply domain/build.gradle]: In this PR no but in the following PR which comes from this I will use it for Junit 5

Comment (Nataniel-Antosik) [reply domain/build.gradle]: Ok it is strange but I tried to delete it in the next PR and still work therefore I will delete it

Comment (Nataniel-Antosik) [reply domain/build.gradle]: I deleted it :D

**Review by MarekMacko — APPROVED** [review]

**Pull request merged**